services/mediastoredata/src/it/java/software/amazon/awssdk/services/mediastoredata/TransferEncodingChunkedIntegrationTest.java

```diff
@@ -0,0 +1,221 @@
+    private static Container container;
+    private static PutObjectRequest putObjectRequest;
+    private static DeleteObjectRequest deleteObjectRequest;
+
+    @BeforeAll
+    public static void setup() {
+        credentialsProvider = getCredentialsProvider();
+        mediaStoreClient = MediaStoreClient.builder()
+                                           .credentialsProvider(credentialsProvider)
+                                           .httpClient(ApacheHttpClient.builder().build())
+                                           .build();
+        container = createContainer();
+        URI uri = URI.create(container.endpoint());
+
+        syncClientWithApache = MediaStoreDataClient.builder()
+                                                   .endpointOverride(uri)
+                                                   .credentialsProvider(credentialsProvider)
+                                                   .httpClient(ApacheHttpClient.builder().build())
+                                                   .overrideConfiguration(o -> o.addExecutionInterceptor(new CaptureTransferEncodingHeaderInterceptor()))
+                                                   .build();
+
+        syncClientWithUrlConnection= MediaStoreDataClient.builder()
+                                                         .endpointOverride(uri)
+                                                         .credentialsProvider(credentialsProvider)
+                                                         .httpClient(UrlConnectionHttpClient.create())
+                                                         .overrideConfiguration(o -> o.addExecutionInterceptor(new CaptureTransferEncodingHeaderInterceptor()))
+                                                         .build();
+
+        asyncClientWithNetty = MediaStoreDataAsyncClient.builder()
+                                                        .endpointOverride(uri)
+                                                        .credentialsProvider(getCredentialsProvider())
+                                                        .httpClient(NettyNioAsyncHttpClient.create())
+                                                        .overrideConfiguration(o -> o.addExecutionInterceptor(new CaptureTransferEncodingHeaderInterceptor()))
+                                                        .build();
+
+        putObjectRequest = PutObjectRequest.builder()
+                                           .contentType("application/octet-stream")
+                                           .path("/foo")
+                                           .build();
+
+        deleteObjectRequest = DeleteObjectRequest.builder()
+                                                 .path("/foo")
+                                                 .build();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        Waiter.run(() -> syncClientWithApache.deleteObject(deleteObjectRequest));
```

Comment:
Don't we need add `.until` to indicate when it should stop and `orFailAfter` to activate it?